source/grid/manifold_lib.cc

```diff
@@ -373,6 +373,104 @@ get_tangent_vector (const Point<spacedim> &p1,
 
 
 
+template<int dim, int spacedim>
+Tensor<1, spacedim>
+SphericalManifold<dim, spacedim>::
+normal_vector (const typename Triangulation<dim, spacedim >::face_iterator &face,
+               const Point<spacedim>                                       &p) const
+{
+  // We assume that the face can either be properly described by a bilinear form
+  // with the vector pointing from the center to p being close to be tangential,
+  // or spherically curved with the normal vector pointing in the direction of
+  // the vector from the center to p. Use a crude approximation of the normal
+  // from the vertices of the face to find out which is the correct choice.
+  const Tensor<1,spacedim> unnormalized_spherical_normal = p-center;
+  const Tensor<1,spacedim> normalized_spherical_normal
+    = unnormalized_spherical_normal/unnormalized_spherical_normal.norm();
+
+  const Tensor<1,spacedim> unnormalized_crude_normal
+    = (spacedim==2) ?
+      cross_product_2d(face->vertex(1)-face->vertex(0)) :
+      cross_product_3d((face->vertex(1)-face->vertex(0)),
+                       (face->vertex(2)-face->vertex(0)));
```

Comment:
To clarify my previous comment, I would have expected that we check that all two/four face vertices are located on the same distance for using the spherical normal. If that is not the case, one can still have the check "not too far away from the `cos(theta)=0.5`" fulfilled, but in that case it will not be correct. Even more, in that case even `flat_manifold.normal_vector()` would not be correct, but it is really the formula with the tangent vectors that should be fulfilled in my opinion.

Reply:
I guess we simply don't have in general a clear definition of the (best) normal vector for faces that don't have a constant distance from the center. SInce also the current implementation (using `get_tangent_vector`) returns the correct normal vector for the boundary of a `*_hyper_shell`, i.e. also for faces that are flat and the tangent plane includes the origin, I am fine with using that one instead of the implementation of `FlatManifold`. 
I don't have a problem with checking for a constant distance from the center instead.